instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/UpcomingRouteObjectsTest.kt

```diff
@@ -0,0 +1,99 @@
+        latitude = 48.143406486859135
+        longitude = 11.428011943347627
+    }
+
+    @Before
+    fun setup() {
+        runOnMainSync {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(activity)
+                    .accessToken(getMapboxAccessTokenFromResources(activity))
+                    .routingTilesOptions(
+                        RoutingTilesOptions.Builder()
+                            .tilesBaseUri(URI(mockWebServerRule.baseUrl))
+                            .build()
+                    )
+                    .build()
+            )
+        }
+    }
+
+    @Test
+    @Ignore("waiting for the NN fix, see NAVAND-1108")
+    fun distanceToIncidentDoNotChangeAfterAddingNewWaypoint() = sdkTest {
+        val routeWithIncident = NavigationRoute.create(
+            directionsResponseJson = readRawFileText(activity, R.raw.route_witn_incident_one_leg),
+            routeRequestUrl = "https://api.mapbox.com/directions/v5/mapbox/driving-traffic/" +
+                "11.428011943347627,48.143406486859135;11.443258702449555,48.14554279886465" +
+                "?access_token=**&alternatives=true" +
+                "&annotations=closure,congestion_numeric,congestion,speed,duration,distance" +
+                "&geometries=polyline6&language=en&overview=full&steps=true",
+            routerOrigin = RouterOrigin.Offboard
+        )
+        val routeWithIncidentTwoLegs = NavigationRoute.create(
+            directionsResponseJson = readRawFileText(activity, R.raw.route_witn_incident_two_legs),
+            routeRequestUrl = "https://api.mapbox.com/directions/v5/mapbox/driving-traffic/" +
+                "11.428011943347627,48.143406486859135;11.42945687746061,48.1436160028498" +
+                ";11.443258702449555,48.14554279886465" +
+                "?access_token=**&alternatives=true" +
+                "&annotations=closure,congestion_numeric,congestion,speed,duration,distance" +
+                "&geometries=polyline6&language=en&overview=full&steps=true",
+            routerOrigin = RouterOrigin.Offboard
+        )
+        val incidentId = "6058002857835914"
+
+        mapboxNavigation.startTripSession()
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routeWithIncident)
+        val upcomingIncidentForOneLeg = mapboxNavigation.routeProgressUpdates()
+            .first { it.currentState == RouteProgressState.TRACKING }
+            .upcomingRoadObjects
+            .first { it.roadObject.id == incidentId }
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routeWithIncidentTwoLegs)
+        val upcomingIncidentForTwoLegsRoute = mapboxNavigation.routeProgressUpdates()
+            .first { it.currentState == RouteProgressState.TRACKING }
+            .upcomingRoadObjects
+            .first { it.roadObject.id == incidentId }
+
+        assertEquals(
```

Comment:
Why do we expect them to be equal?
The routes are different, do they have the same geometry (at least until the incident)? If yes, it should be specified in the comments. If the geometry is different, `distanceToStart` may be different (it's distance along the route, not direct distance, right?).

Reply:
> Why do we expect them to be equal?

Current location doesn't change and both routers heads to the same direction thought the same incident.

Reply:
> If yes, it should be specified in the comments.

Thank you for the feedback. I will try to update the test

Reply:
@dzinad , I updated the test. Do you understand what and why it checks now?

Reply:
My point is that if we have the same origin and the same incident location, when we add a waypoint, distances may still differ. See the drawing.
And it's still not clear that the geometries from the origin until the incident are exactly the same.
Or am I missing something?
![telegram-cloud-photo-size-2-5375224774822839390-y](https://user-images.githubusercontent.com/32109537/214827866-466dcd08-5166-40a9-8c14-ecca364a290a.jpg)

Reply:
I see your point, thank you. The routes I test follow the same roads. Like 
![image](https://user-images.githubusercontent.com/6190346/214850939-9d445c80-a183-476f-9759-733c27d6ad8e.png)